### PR TITLE
Swaggerize

### PR DESCRIPTION
**Pull request by dshiga**
Added swagger annotations to SnoopApiService and SnoopApiFormats. Had to stop using the ~ operator to chain routes together, because swagger needs a separate variable for each route, so it has something to hang the annotations on.

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/snoop/ws/SnoopApiService.scala [hunk 1]
```diff
@@ -17,21 +23,54 @@ object SnoopApiServiceActor {
...
-  def receive = runRoute(snoopRoute)
-}
+  def possibleRoutes = baseRoute ~ workflowRoutes ~ swaggerService.routes 
 
+  def receive = runRoute(possibleRoutes)
 
-// this trait defines our service behavior independently from the service actor
-trait SnoopApiService extends HttpService {
-  implicit def executionContext = actorRefFactory.dispatcher
+  private val snoopConfig: Config = ConfigFactory.parseFile(new File("/etc/snoop.conf"))
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/snoop/ws/SnoopApiService.scala]: I don't want to be looking up config here. Please pass the config in via the constructor and lookup the config in Boot.scala. You can pass in a swaggerConfig object that I would suggest be a case class constructed from Config loaded in Boot.

File: src/main/scala/org/broadinstitute/dsde/snoop/ws/SnoopApiService.scala [hunk 2]
```diff
@@ -17,21 +23,54 @@ object SnoopApiServiceActor {
...
+        swaggerConfig.getString("termsOfServiceUrl"),
+        swaggerConfig.getString("contact"),
+        swaggerConfig.getString("license"),
+        swaggerConfig.getString("licenseUrl"))
+    )
+  }
+}
+
+@Api(value = "", description = "Snoop Base API", position = 1)
+trait BaseSnoopApiService extends HttpService {
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/snoop/ws/SnoopApiService.scala]: I would call this Root, not Base. The latter makes me think that something extends this.

File: src/main/scala/org/broadinstitute/dsde/snoop/ws/SnoopApiService.scala [hunk 3]
```diff
@@ -49,7 +88,29 @@ trait SnoopApiService extends HttpService {
...
+
+// this trait defines our service behavior independently from the service actor
+@Api(value = "workflowExecutions", description = "Snoop Workflow Execution API", position = 1)
+trait WorkflowExecutionApiService extends HttpService {
+  import WorkflowExecutionJsonSupport._
+  import SprayJsonSupport._
+
+  def executionServiceHandler: RequestContext => WorkflowExecutionService
+
+  def workflowRoutes = startWorkflowRoute ~ workflowStatusRoute
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/snoop/ws/SnoopApiService.scala]: does it work if you put the cookie directive up here? That way you don't need to duplicate the cookie extraction.

File: src/main/scala/org/broadinstitute/dsde/snoop/ws/SnoopApiService.scala [hunk 4]
```diff
@@ -17,21 +23,54 @@ object SnoopApiServiceActor {
...
+  def receive = runRoute(possibleRoutes)
 
-// this trait defines our service behavior independently from the service actor
-trait SnoopApiService extends HttpService {
-  implicit def executionContext = actorRefFactory.dispatcher
+  private val snoopConfig: Config = ConfigFactory.parseFile(new File("/etc/snoop.conf"))
+  private val swaggerConfig = snoopConfig.getConfig("swagger")
+
+  val swaggerService = new SwaggerHttpService {
+    override def apiTypes = Seq(typeOf[BaseSnoopApiService], typeOf[WorkflowExecutionApiService])
```
Comment (geoffjentry) [inline src/main/scala/org/broadinstitute/dsde/snoop/ws/SnoopApiService.scala]: Prefer using Vector as your concrete instantiation of Seq. Seq() ends up creating a List but Vector is generally more performant

**Pull request merged**